src/Command/RectorRoutesExportCommand.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Symfony\Command;
+
+use Symfony\Bundle\FrameworkBundle\Console\Helper\DescriptorHelper;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * This command must be added to the application to export the routes to work with the "AddRouteAnnotationRector" rule
```

Comment:
There exists some files with information about routes inside the `var/cache` directory but the information is not direct accessible and the controllers are not resolved. There still stand `FooBundle:Bar:baz`.

So I tried to use the symfony command `debug:router --format=json` but there are also the controllers not resolved. But if you use this command inside the application it resolves the controllers correct.

TBD

Reply:
Could work but I'm failing :/ Maybe that can do someone smarter than me

Reply:
I see. Thanks for trying all the options. We can merge this PR, if CI is passing. The code does not have to be complete.

Could you share a minimal reproducible repository with your setup?
I see what can we do.

Reply:
I've created a minimal Symfony 3.4 application which should help to debug.
https://github.com/malteschlueter/symfony-reproducers/tree/feature/rector-add-route-annotation

That is what I've changed in Rector to require the container
```diff
diff --git a/src/ValueObjectFactory/RouteMapFactory.php b/src/ValueObjectFactory/RouteMapFactory.php
index 8adee63..5681cbc 100644
--- a/src/ValueObjectFactory/RouteMapFactory.php
+++ b/src/ValueObjectFactory/RouteMapFactory.php
@@ -17,6 +17,8 @@ final class RouteMapFactory
 
     public function createFromFileContent(string $configFilePath): RouteMap
     {
+        require $configFilePath;
+
         $fileContents = $this->smartFileSystem->readFile($configFilePath);
         $json = json_decode($fileContents, true);
 
diff --git a/tests/Rector/ClassMethod/AddRouteAnnotationRector/config/configured_rule.php b/tests/Rector/ClassMethod/AddRouteAnnotationRector/config/configured_rule.php
index 8fbc08d..e98921c 100644
--- a/tests/Rector/ClassMethod/AddRouteAnnotationRector/config/configured_rule.php
+++ b/tests/Rector/ClassMethod/AddRouteAnnotationRector/config/configured_rule.php
@@ -8,5 +8,5 @@ use Rector\Symfony\Rector\ClassMethod\AddRouteAnnotationRector;
 return static function (RectorConfig $rectorConfig): void {
     $rectorConfig->import(__DIR__ . '/../../../../../config/config.php');
     $rectorConfig->rule(AddRouteAnnotationRector::class);
-    $rectorConfig->symfonyRoutesJson(__DIR__ . '/route.json');
+    $rectorConfig->symfonyRoutesJson('/Users/mschlueter/Projects/Privat/symfony-reproducers/var/cache/dev/AppDevDebugProjectContainer.php');
 };
```

Reply:
PHPStan says it doesn't know `Symfony\Component\Routing\RouterInterface::getRouteCollection()` but the component is installed. See https://github.com/rectorphp/rector-symfony/runs/6442259714?check_suite_focus=true#step:4:124
How does I solve it?

Unfortunately I've no idea what this job is doing and why it's failing :D Any idea? https://github.com/rectorphp/rector-symfony/runs/6442259666?check_suite_focus=true

Reply:
> Unfortunately I've no idea what this job is doing and why it's failing :D Any idea?

Here it's Github UX fail. Check the 2 jobs above, this one is changing code:

```bash
vendor/bin/rector process --ansi
```

Best way to fix is running it locally and commit :)

Reply:
And thank you very much for your help @TomasVotruba :)
Hope we can finish this feature and it works well for the most people (everyone will take some time I think :D )
I think it can help many people who works on legacy applications and wants to update to annotation/attributes

Reply:
> PHPStan says it doesn't know Symfony\Component\Routing\RouterInterface::getRouteCollection() but the component is installed. See [rectorphp/rector-symfony/runs/6442259714?check_suite_focus=true#step:4:124](https://github.com/rectorphp/rector-symfony/runs/6442259714?check_suite_focus=true#step:4:124)
How does I solve it?


You can skip this file in `phpstan.neon`, as it requires external context

Reply:
> And thank you very much for your help @TomasVotruba :)

You're welcome :+1:  

I have no doubts, we'll finish it this week to test it out on edge-cases. You've done great works so far, thank you :bow:

Reply:
I'm in the https://github.com/malteschlueter/symfony-reproducers/tree/feature/rector-add-route-annotation in `rector-add-route-annotation` branch. But when I run:

```bash
vendor/bin/rector
```

it's missing the package. What exact steps should I do to make Rector fail?

Reply:
Btw, after Rector + PHPStan fix in CI, we can merge it to progress with testing on `rector/rector-dev-main` :wink: